### DISCUSSION TITLE
Need help with an experiment

### DISCUSSION THREAD

**Discussion by Karma-Tron** [official-stockfish/Stockfish#4038] [Q&A]
I'm trying to create a version where each move is searched as full root moves, aka MultiPV, but still output only the best PV of each iteration. In search.cpp, I can change statement 2 in the line below, but then all PV's are being displayed.

 `      // MultiPV loop. We perform a full root search for each PV line `
 `      for (pvIdx = 0; pvIdx < multiPV && !Threads.stop; ++pvIdx) `

How can I suppress the extra PV's in the output or (probably even better) search all child nodes  as full root moves?

Thanks.

**Comment by ghost** [comment]
the best way to search all move as full root is create loop in uci.cpp

```
for (MoveList<LEGAL>(pos)) {
position(pos, is, states);
go(pos, is, states);
}
```

**Comment by Karma-Tron** [comment]
Thanks, but I was not able to get this working.

**Reply by ghost** [reply]
do you want complete code for uci.cpp ? its very easy , You will be able to to working it

**Comment by Karma-Tron** [comment]
That would be too much to ask probably, but perhaps I might be able to figure it out with some extra clues.

I had experience with Basic (who doesn't) 40 years ago, and professionally with dBase and Thomson Assembly from 25-30 years ago, but that's about it and most of it rusted away over the years.

My programming experience with C++ is very basic.

**Comment by Sopel97** [comment]
Change `UCI::pv` in search.cpp to not iterate over multipvs

**Comment by Karma-Tron** [comment]
Thank you, that makes sense. I'll give it a whirl.

**Comment by Karma-Tron** [comment]
Alas, I could not make it work and gave up.
